utils/gear-wasmer-cache/src/lib.rs

```diff
@@ -167,7 +167,12 @@ impl FileSystemCache {
     fn load(&self, key: Hash) -> Result<Vec<u8>, io::Error> {
         let path = self.path.join(key.to_string());
 
-        fs::read(path)
+        let mut file = File::open(path)?;
+        file.lock_exclusive()?;
```

Comment:
Is  `unlock` called  on drop?

Reply:
> Furthermore, the lock is released either by an explicit LOCK_UN operation on any of these duplicate descriptors, or when all such descriptors have been closed.

https://linux.die.net/man/2/flock

Reply:
Explicit unlock mb make sense for windows
> If a process terminates with a portion of a file locked or closes a file that has outstanding locks, the locks are unlocked by the operating system. However, the time it takes for the operating system to unlock these locks depends upon available system resources.

https://learn.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-lockfileex